roles/common/tasks/networking.yml

```diff
@@ -13,6 +13,36 @@
   with_items: network_interfaces_d
   tags: ['prereboot']
 
+- name: set gateway address for multi subnet 
+  set_fact: dfl_nw_gw="{{ item.gateway }}"
+  when: (update_routes|default('false')|bool) and ((ansible_default_ipv4.network ==  ( item.cidr | ipaddr('network') ))) 
+  with_items: "{{ undercloud_cidr }}"
+
+- name: add static route for multi subnet
+  command: "/sbin/ip route replace {{ item.cidr }} via {{ dfl_nw_gw }}"
+  when: (update_routes|default('false')|bool) and (not (ansible_default_ipv4.network ==  ( item.cidr | ipaddr('network') )))
+  with_items: "{{ undercloud_cidr }}"
+
+- name: Check if interface has its own config file
+  stat: path=/etc/network/interfaces.d/{{ ansible_default_ipv4.interface }}.cfg
+  register: found_in_interfaces_d 
+  when: (update_routes|default('false')|bool)
+
+- name: Set network config file
+  set_fact: nw_config_file="/etc/network/interfaces.d/{{ ansible_default_ipv4.interface }}.cfg"
+  when: (update_routes|default('false')|bool) and found_in_interfaces_d.stat.exists is defined and found_in_interfaces_d.stat.exists
+
+- name: Set network config file
+  set_fact: nw_config_file="/etc/network/interfaces"
+  when: (update_routes|default('false')|bool) and ( not found_in_interfaces_d.stat.exists|bool )
+
+- name: Check for static routes in {{ nw_config_file }}
+  command: "grep -h 'ip route add {{ item.cidr }} via {{ dfl_nw_gw }}' {{ nw_config_file }}"
+  with_items: "{{ undercloud_cidr }}"
+  always_run: True
```

Comment:
I was more confused about the use of `always_run`. That is a task control that determines whether or not a task will be done when running `ansible-playbook` in "check mode" (where it doesn't do the tasks, it just says what it _would_ do). We don't really support check mode anywhere, so I was curious why it was being considered here.